controllers/topology/desired_state.go

```diff
@@ -249,6 +254,11 @@ func computeMachineDeployment(_ context.Context, s *scope.Scope, machineDeployme
 		currentObjectRef:      currentInfraMachineTemplateRef,
 	})
 
+	// Add ClusterTopologyMachineDeploymentLabel to the generated InfrastructureMachine template
+	infraMachineTemplateLabels := map[string]string{}
```

Comment:
Same as above (let's keep the labels from desiredMachineDeployment and initialise them to empty map only if nil)